### ISSUE TITLE
MobileNetV3 Large Training All From Scratch

### ISSUE THREAD

**Issue by whysetiawan** [leondgarse/Keras_insightface#109]
Hi Thanks for providing this tutorial [All from scratch](https://github.com/leondgarse/Keras_insightface/discussions/71). I have been following step by step and using Casia WebFace Datasets and the training progress increasing so slow. 

Here is my accuracy result
![Uploading Screenshot 2023-04-02 at 18.14.35.png…](Accuracy)

First i was using LR: 0.0001 and change to 0.001 at 25 epochs in order to improve accuracy. It didn't help, i wonder if you have advice

**Comment by whysetiawan** [comment]
My Source code is here [](https://www.kaggle.com/code/whysetiawan27/arcface-mobilenetv3large) version 16 (before i make changes)

**Comment by leondgarse** [comment]
Just took a quick scan, great job taking it to kaggle!
- You may try a smaller `scale=16.0` for `arcface` as a start, larger one may hard to converge.
- For `SGD`, you may start with a larger lr like `0.1`.
- Another point is that, in my very early tests, `weight_decay` is crucial for `arcface` loss. You may add `l2_regularizer` to model, or use optimizers like `tfa.optimizers.SGDW` or `tfa.optimizers.AdamW`.

May try some tests myself if got some time, but not sure...

**Comment by whysetiawan** [comment]
Thank s for your response. I'll make some changes and will be back if face another problem. Just a question, how many epochs usually take to train from scratch? I read [Training Scripts](https://github.com/leondgarse/Keras_insightface/discussions/15) and your result is always high at the first epochs.

**Comment by leondgarse** [comment]
Ya, For MS1MV3 dataset, it should give some reasonable result after first expoch. Not sure for CASIA datasets, as it's much smaller, but shouldn't be larger than 3 epochs.

**Comment by aidansmyth95ifx** [comment]
Any luck training this model? It boasts great memory and latency characteristics, curious what its performance looks like vs the other models.
